### ISSUE TITLE
add_method_tracer modifies object arguments

### ISSUE THREAD

**Issue by dipth** [newrelic/newrelic-ruby-agent#424]
`add_method_tracer` causes object arguments to be cloned.

## Description
When tracing a method that takes an object argument that is supposed to be mutated during the execution of that method, the tracing causes the passed argument to be a clone of the object, meaning that any mutations are lost to outside references to the same object.

## Expected Behavior
No cloning of arguments should happen and the passed object should be included in the call to the method as is.

## Steps to Reproduce
```ruby
class Demonstration
  include ::NewRelic::Agent::MethodTracer

  def call
    (1..10).each_with_object({}) do |i, hash|
      process(i, hash)
    end
  end

  def process(i, hash)
    hash[i] = i * 2
  end

  add_method_tracer :process, 'Demonstration/process'
end

Demonstration.new.call
#=> {}
```

Expected result:

```ruby
Demonstration.new.call
=> {1=>2, 2=>4, 3=>6, 4=>8, 5=>10, 6=>12, 7=>14, 8=>16, 9=>18, 10=>20}
```

## Your Environment
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19]

## Additional context
It seems like this is not an issue in Ruby 2.6

**Comment by tannalynn** [comment]
Hello @dipth. We will take a closer look at this.
Thank you for bringing this to our attention.

**Comment by mwlang** [comment]
This should be solved by this PR: #422
